File: app/models/candidat.py
from dataclasses import dataclass
from typing import Optional


@dataclass
class Candidat:
    candidat_id: int = 0
    nom: str = ""
    nom_intermediaire: str = ""
    prenom: str = ""
    date_de_naissance: Optional[str] = None  # stored as TEXT "YYYY-MM-DD"
    sexe: str = ""
    candidat_statut: str = "I"               # "I" = interne, "E" = externe
    langue: str = "fr"
    etab_id: int = 0
    candidat_moyenne: Optional[float] = None
    candidat_classement: Optional[int] = None
    anonymat: str = ""

File: OrientationPFIEV/Models/Candidat.cs
namespace OrientationPFIEV.Models;

/// <summary>
/// Exam candidate. CandidatStatut: "I" = internal, "E" = external.
/// Langue: "fr" or "vi". Sexe: "M" or "F".
/// </summary>
public class Candidat
{
    public int CandidatID { get; set; }
    public string Nom { get; set; } = "";
    public string NomIntermediaire { get; set; } = "";
    public string Prenom { get; set; } = "";
    public DateTime? DateDeNaissance { get; set; }
    public string Sexe { get; set; } = "";
    public string CandidatStatut { get; set; } = "";
    public string Langue { get; set; } = "";
    public int EtabID { get; set; }
    public double? CandidatMoyenne { get; set; }
    public int? CandidatClassement { get; set; }
    public string Anonymat { get; set; } = "";
}
